doc/howto/write_an_alphabet/index.md

```diff
@@ -0,0 +1,150 @@
+~~~
+
+## Prerequisites
+
+A look at the documentation of seqan3::Alphabet will reveal that it is actually a refinement of other concepts, 
+more precisely seqan3::Semialphabet which in turn refines std::CopyConstructible and std::StrictTotallyOrdered. 
+Let's check those:
+
+~~~cpp
+static_assert(std::CopyConstructible<dna2>);    // OK
+static_assert(std::StrictTotallyOrdered<dna2>); // compiler error
+static_assert(seqan3::Semialphabet<dna2>);      // compiler error
+static_assert(seqan3::Alphabet<dna2>);          // compiler error
+~~~
+
+You should see that your type models only the std::CopyConstructible concept.
+Let's have a look at the documentation for std::StrictTotallyOrdered. Can you guess what it describes?
+
+It describes the requirements for types that are comparable via `<`, `<=`, `>` and `>=`.
+This is useful so that you can sort a text over the alphabet, for example.
+Additionally, in order to model std::StrictTotallyOrdered we need to make our alphabet
+[equality comparable](\ref std::EqualityComparable) by implementing the (in-)equality operators (as free functions).
+\snippet alphabet_custom_steps.cpp equality
+
+\assignment{Excercise}
+Implement the inequality operator (!=) for `dna2`.
+\endassignment
+\solution
+\snippet alphabet_custom_steps.cpp inequality
+\endsolution
+
+We see that our type now models std::EqualityComparable, which is a prerequisite of std::StrictTotallyOrdered.
+\snippet alphabet_custom_steps.cpp equality_comparable
+
+\assignment{Excercise}
+Implement the four compare operators *as free functions* and verify that your type models std::StrictTotallyOrdered.
+\endassignment
+\solution
+\snippet alphabet_custom_steps.cpp comparison
+\endsolution
+
+## Semialphabet
+
+Let's move on to the more interesting concepts. seqan3::Semialphabet requires the *rank* interface
+that we have introduced in the [alphabet tutorial](\ref tutorial_alphabets). If we define the member functions
+`to_rank`, `assign_rank` and `alphabet_size` for our type, SeqAn automatically exposes them to the respective global
+functions `seqan3::to_rank`, `seqan3::assign_rank_to` and `seqan3::alphabet_size`. 
+Note that `assign_rank` is only required for seqan3::WritableSemialphabet.
+\snippet alphabet_custom_steps.cpp writable_semialphabet
+
+## Alphabet
+
+Now that you have a feeling for concepts, have a look at seqan3::Alphabet and make your type 
+also model the seqan3::Alphabet concept. If you keep the suggested namings, SeqAn will automatically expose
+the members to the respective global functions that model the concept.
+
+\assignment{Excercise}
+Please implement the member function `to_char(void)` which returns the char 'S' or 'W'
+depending on the rank value.
+\endassignment
+\solution
+\snippet alphabet_custom_steps.cpp alphabet
+\endsolution
+
+\assignment{Excercise}
+Now model the full seqan3::WritableAlphabet concept.
+
+1. Implement the member function `dna2 & assign_char(char const)`.
```

Comment:
I would put the `assign char` task together with `to char` and then write one more sentence

```
seqan3::Alphabet additionally requires that the type has a member function `bool char_is_valid` which can be used to determine whether a given character is an actual representation of the alphabet. For `dna2`, `S` and `W`, as well as `s` and `w`,  should be valid but nothing else. 
```

before the write `char_is_valid` task